crates/holochain_data/src/kind.rs

```diff
@@ -0,0 +1,133 @@
+//! Database kind types for identifying different database instances.
+//!
+//! This module provides typed identifiers for the different kinds of databases
+//! used in Holochain. Each kind implements [`DatabaseIdentifier`] to provide
+//! a unique string identifier for the database file.
+
+use crate::DatabaseIdentifier;
+use holo_hash::{AgentPubKey, DnaHash};
+use holochain_zome_types::cell::CellId;
+use std::sync::Arc;
+
+/// Specifies the database used for authoring data by a specific cell.
+///
+/// Each cell (DNA/Agent pair) has its own authored database.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Authored {
+    cell_id: Arc<CellId>,
+    id: String,
+}
+
+impl DatabaseIdentifier for Authored {
+    fn database_id(&self) -> &str {
+        &self.id
+    }
+}
+
+impl Authored {
+    /// Create a new authored database identifier for a cell.
+    pub fn new(cell_id: Arc<CellId>) -> Self {
+        let id = format!("authored-{}-{}", cell_id.dna_hash(), cell_id.agent_pubkey());
+        Self { cell_id, id }
+    }
+
+    /// Get the DNA hash for this authored database.
+    pub fn dna_hash(&self) -> &DnaHash {
+        self.cell_id.dna_hash()
+    }
+
+    /// Get the agent public key for this authored database.
+    pub fn agent_pubkey(&self) -> &AgentPubKey {
+        self.cell_id.agent_pubkey()
+    }
+}
+
+/// Specifies the database used for DHT data for a specific DNA.
+///
+/// All cells on the same DNA share the same DHT database.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Dht {
```

Comment:
Dht and Authored are not needed for the WASM database. I suggest we leave them for subsequent PRs.

When its relevant, I would still like to advocate for no longer having a separate Authored database.

Reply:
Without private entry data I would agree, but I don't have a clear idea of how we'd go about refactoring all code that accesses the DHT, when it's currently only queries against the authored database that are really careful about not leaking private entry data.